search/includes/classes/class-search.php

```diff
@@ -112,6 +112,16 @@ public function init() {
 		$this->setup_regular_stat_collection();
 	}
 
+	/**
+	 * Check if the constants needed for ElasticSearch connection are defined.
+	 */
+	public static function are_es_constants_defined() {
+		$endpoints_defined = defined( 'VIP_ELASTICSEARCH_ENDPOINTS' );
```

Comment:
We should also check that this is a non-empty array.

Reply:
Good point!